source/fundamentals/connection/connection-options.txt

```diff
@@ -6,203 +6,290 @@ Connection Options
 
-   * - **minPoolSize**
-     - integer
-     - ``0``
-     - Specifies the minimum number of connections that must exist at any moment
-       in a single connection pool.
+   * - **readConcernLevel**
+     - string
+     - ``null``
+     - Specifies the default read concern for the client. See :ref:`read concern
+       <read-concern>` for more information.
 
-   * - **noDelay**
-     - boolean
-     - ``true``
-     - Specifies whether to use the TCP socket no-delay option. For more
-       information, see the documentation for `Node.js socket.setNoDelay
-       <https://nodejs.org/dist/latest-v10.x/docs/api/net.html#net_socket_setnodelay_nodelay>`_.
+   * - **readPreference**
+     - string
+     - ``"primary"``
+     - Specifies the default read preference for the client (excluding tags). See :ref:`read preference <read-preference>` for more information.
 
-   * - **pkFactory**
-     - object
+   * - **readPreferenceTags**
+     - comma-separated key:value pairs (e.g. "dc:ny,rack:1" and "dc:ny)
+       can be specified multiple times, each instance of this key is a
+       separate tag set
      - ``null``
-     - Specifies a primary key factory object that generates custom ``_id``
-       keys.
+     - Specifies the default read preference tags for the client. This option is
+       valid only if the read preference mode is not primary
 
-   * - **promiseLibrary**
-     - object
+       The order of the tag sets in the read preference is the same as the order
+       they are specified in the URI.
+
+   * - **replicaSet**
+     - string
      - ``null``
-     - Specifies the Promise library class the application uses (e.g. Bluebird).
-       This library must be compatible with ES6.
+     - Specifies the name of the replica set to connect to.
 
-   * - **promoteBuffers**
+   * - **retryReads**
      - boolean
-     - ``false``
-     - Specifies whether to promote Binary BSON values to native Node.js
-       ``Buffer`` type data.
+     - ``true``
+     - Enables retryable reads on MongoDB version 3.6 or later.
 
-   * - **promoteLongs**
+   * - **retryWrites**
      - boolean
      - ``true``
-     - Specifies whether to convert ``Long`` values to a number if they fit
-       inside 53 bits of resolution.
+     - Enables retryable writes on MongoDB version 3.6 or later.
+
+   * - **serverSelectionTimeoutMS**
+     - non-negative integer
+     - ``30000``
+     - Specifies the timeout, in milliseconds, to block for server selection before raising an error.
 
-   * - **promoteValues**
+   * - **serverSelectionTryOnce**
      - boolean
      - ``true``
-     - Specifies whether to promote BSON values to Node.js native types when
-       possible. When set to false, it uses wrapper types to present
-       BSON values.
+     - Scan the topology only once after a server selection failure instead of repeatedly until the server selection times out
 
-   * - **raw**
+   * - **socketTimeoutMS**
+     - non-negative integer
+     - ``0``
```

Comment:
Issue: Could the meaning of the default value be stated explicitly? I believe 0 means no timeout.